test/unit_tests/braket/experimental/autoqasm/test_operators.py

```diff
@@ -794,29 +791,44 @@ def test_assignment_py(value: Any) -> None:
 @pytest.mark.parametrize("value", [0, 1])
 def test_py_if_stmt(value: int) -> None:
     """Test Python if branches on true and false conditions."""
+    # todo: if statements are no longer optimized out during binding parameters
+    # we will need to be explicit about this compilation pass functionality
+    # we can still add it to `make_bound_program`
```

Comment:
Before, users would define a "Program builder" with parameters that get bound during the function call:
```
@aq.main
def my_program_builder(builder_input: int):
    ...

my_program = my_program_builder(builder_input=3)
```
We still expose that functionality by allowing customers to create builder methods, but customers who simply want to make a program with arguments will likely transition to the following:
```
@aq.main
def my_program(program_input: int):
    ...

my_program = my_program(program_input=3)
```
In this case, "binding" the arguments happens in a different place and has a slightly different effect than before (creates `input` variables vs inlines the variables). My comment is just saying we should be aware of this and intentional about providing an option to "inline" a variable of a Program if we want to expose that functionality to customers outside of meta-programming.

"optimizing out if statements" is one example of a simplification that may occur while inlining a variable

Reply:
Can you make a new issue on the board with the change you desire? Please focus the title of the issue on the action that should be completed by the issue. This reads more like a statement: `if statements are no longer optimized out during binding parameters`

Since `if` statement is just one example of this behavior, let's remove this comment. The action taken is likely more broad than this one test or operator, and an issue on our board will be much more effective to call our attention to this while we are planning.

Reply:
Yes, I will remove this comment, since it is no longer relevant once I switch this test back to using an in-scope python variable for `a` instead of an input. I've captured the general todo here in https://github.com/amazon-braket/amazon-braket-sdk-python/issues/815